solrmonitor/zkwatcherman.go

```diff
@@ -175,8 +178,13 @@ func (m *zkWatcherMan) monitorChildren(synch bool, path string, onchange func(ch
 			}
 			return nil
 		}
-		cont := onchange(children)
-		if cont {
+		cont, err := onchange(children)
+		if err != nil {
+			// We failed to call the onchange handler; add a task for the recovery thread to keep trying
+			m.logger.Printf("zkWatcherMan %s: error calling onchange: %s", path, err)
+			m.enqueueDeferredTask(deferredChildrenTask{path: path, onchange: onchange})
```

Comment:
Could this get stuck in an error loop and spam the recovery thread? Should enqueue here check a max queue length and then bail with a error log if it's over?

Reply:
I don't think the queue would grow unbounded. If we reach this point, the original `getChildrenAndWatch` already succeeded and the corresponding `monitorChildren` returned no error meaning the deferred task would be removed from the queue. Further errors would result in a new deferred task, but no ongoing watch. Basically a given collection should either have an ongoing watch or a deferred task, but never both.